### ISSUE TITLE
Allow github to use SSH protocol for clones even when project key not present

### ISSUE THREAD

**Issue by fbenas** [php-censor/php-censor#398]
I wish to always use the SSH protocol for every project. I am using an SSH key that is set in the user's home directory at `~/.ssh/` that allows git to clone from GitHub; authentication like this seems reasonable and works correctly.

However, because of the check in the `GithubBuild.php` class https://github.com/php-censor/php-censor/blob/master/src/Model/Build/GithubBuild.php#L173, HTTP will be forced when there's no private key present at the project level.

Could it perhaps be possible to add another check here for some application level config that forces us to the SSH protocol regardless of project level settings?

Another option here could be to use the Git build type instead. The downside of this method, is we then lose the ability to get build updates in Github from php-censor. This is not ideal as it would be preferred to have all the features of the GithubBuild regardless of the authentication or cloning method.

**Comment by corpsee** [comment]
@fbenas I totally agree with your vision. And I have plans to refactor the build sources. But I don't know how soon.
